## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#7df5207b-139c-4317-89b5-039acdb6b5c0) |
| mint-dist-erasure.sh | :x: [more...](#b5d406bc-d1ae-46c5-8f7e-8d5b0bdd74fa) |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#9a745339-25e7-400b-afe5-8b8a5f8e0548) |
| mint-pools.sh | :x: [more...](#8bcf9998-84f5-49a9-a745-689a02471672) |
<a name="7df5207b-139c-4317-89b5-039acdb6b5c0"></a>
### 13941-f8a3fe9/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s4:32199
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 44 seconds
(4/15) Running aws-sdk-ruby tests ... done in 5 seconds
(5/15) Running awscli tests ... done in 1 minutes and 52 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 25 seconds
(8/15) Running minio-dotnet tests ... done in 51 seconds
(9/15) Running minio-go tests ... done in 1 minutes and 13 seconds
(10/15) Running minio-java tests ... FAILED in 1 minutes and 29 seconds
{
  "name": "minio-java",
  "function": "deleteUser()",
  "duration": 60004,
  "status": "FAIL",
  "error": "java.net.SocketTimeoutException: timeout >>> [okio.SocketAsyncTimeout.newTimeoutException(JvmOkio.kt:143), okio.AsyncTimeout.access$newTimeoutException(AsyncTimeout.kt:162), okio.AsyncTimeout$source$1.read(AsyncTimeout.kt:340), okio.RealBufferedSource.indexOf(RealBufferedSource.kt:449), okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.kt:333), okhttp3.internal.http1.HeadersReader.readLine(HeadersReader.kt:29), okhttp3.internal.http1.Http1ExchangeCodec.readResponseHeaders(Http1ExchangeCodec.kt:178), okhttp3.internal.connection.Exchange.readResponseHeaders(Exchange.kt:106), okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.kt:79), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:34), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201), okhttp3.internal.connection.RealCall.execute(RealCall.kt:154), io.minio.admin.MinioAdminClient.execute(MinioAdminClient.java:173), io.minio.admin.MinioAdminClient.deleteUser(MinioAdminClient.java:266), TestMinioAdminClient.deleteUser(TestMinioAdminClient.java:136), TestMinioAdminClient.runAdminTests(TestMinioAdminClient.java:148), FunctionalTest.runTests(FunctionalTest.java:3720), FunctionalTest.runEndpointTests(FunctionalTest.java:3818), FunctionalTest.main(FunctionalTest.java:3938)]"
}
(10/15) Running minio-js tests ... FAILED in 19 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 25,
  "status": "FAIL",
  "error": "S3Error: Resource requested is unreadable, please reduce your request rate at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:89:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:164:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:88:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1334:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(10/15) Running minio-py tests ... FAILED in 8 minutes and 7 seconds
{
  "name": "minio-py:test_list_buckets",
  "status": "FAIL",
  "message": "S3 operation failed; code: BucketAlreadyOwnedByYou, message: Your previous request to create the named bucket succeeded and you already own it., resource: /minio-py-test-0af4f600-9234-407a-87ac-809937809f83, request_id: 16C1A1C8BF7D6CF0, host_id: d6b2724c-bb6a-4a93-ad33-7716c4565648, bucket_name: minio-py-test-0af4f600-9234-407a-87ac-809937809f83",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 243, in test_list_buckets\n    _call(log_entry, _CLIENT.make_bucket, bucket_name)\n  File \"/mint/run/core/minio-py/tests.py\", line 109, in _call\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 612, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 380, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketAlreadyOwnedByYou, message: Your previous request to create the named bucket succeeded and you already own it., resource: /minio-py-test-0af4f600-9234-407a-87ac-809937809f83, request_id: 16C1A1C8BF7D6CF0, host_id: d6b2724c-bb6a-4a93-ad33-7716c4565648, bucket_name: minio-py-test-0af4f600-9234-407a-87ac-809937809f83\n",
  "function": "make_bucket(bucket_name, location, object_lock)",
  "duration": 241444
}
(10/15) Running s3cmd tests ... done in 38 seconds
(11/15) Running s3select tests ... FAILED in 12 minutes and 36 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "bucket_name": "s3select-test-5c0c2bbb-98f7-4b8c-a744-3435fcde4400"
  },
  "duration": 755725,
  "message": "S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /s3select-test-5c0c2bbb-98f7-4b8c-a744-3435fcde4400, request_id: 16C1A281A6060234, host_id: d6b2724c-bb6a-4a93-ad33-7716c4565648, bucket_name: s3select-test-5c0c2bbb-98f7-4b8c-a744-3435fcde4400",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/s3select/csv.py\", line 39, in test_sql_api\n    data = client.select_object_content(bucket_name, object_name, sql_opts)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 563, in select_object_content\n    response = self._execute(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 397, in _execute\n    return self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 380, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: SlowDown, message: Resource requested is unreadable, please reduce your request rate, resource: /s3select-test-5c0c2bbb-98f7-4b8c-a744-3435fcde4400/c72e4a8c-4e8f-48b9-96b4-1cf5cdd330ce, request_id: 16C1A22D9056E522, host_id: d6b2724c-bb6a-4a93-ad33-7716c4565648, bucket_name: s3select-test-5c0c2bbb-98f7-4b8c-a744-3435fcde4400, object_name: c72e4a8c-4e8f-48b9-96b4-1cf5cdd330ce\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/s3select/csv.py\", line 117, in test_csv_input_custom_quote_char\n    test_sql_api(f'test_{idx}', client, bucket_name,\n  File \"/mint/run/core/s3select/csv.py\", line 47, in test_sql_api\n    raise ValueError(\nValueError: Test test_8 unexpectedly failed with: S3 operation failed; code: SlowDown, message: Resource requested is unreadable, please reduce your request rate, resource: /s3select-test-5c0c2bbb-98f7-4b8c-a744-3435fcde4400/c72e4a8c-4e8f-48b9-96b4-1cf5cdd330ce, request_id: 16C1A22D9056E522, host_id: d6b2724c-bb6a-4a93-ad33-7716c4565648, bucket_name: s3select-test-5c0c2bbb-98f7-4b8c-a744-3435fcde4400, object_name: c72e4a8c-4e8f-48b9-96b4-1cf5cdd330ce\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"./tests.py\", line 53, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 120, in test_csv_input_custom_quote_char\n    client.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 669, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 397, in _execute\n    return self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 380, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /s3select-test-5c0c2bbb-98f7-4b8c-a744-3435fcde4400, request_id: 16C1A281A6060234, host_id: d6b2724c-bb6a-4a93-ad33-7716c4565648, bucket_name: s3select-test-5c0c2bbb-98f7-4b8c-a744-3435fcde4400\n",
  "status": "FAIL"
}
(11/15) Running versioning tests ... FAILED in 12 minutes and 55 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-t90qszwssjcqly",
    "expiry": 60000000000,
    "objectName": "testObject"
  },
  "duration": 126674,
  "error": "want {versions:[{name:dir/ isLatest:true} {name:dir/object isLatest:true}] commonPrefixes:[]}, got {versions:[{name:dir/ isLatest:true} {name:dir/object isLatest:true} {name:dir/object isLatest:false}] commonPrefixes:[]}",
  "function": "/mint/build/versioning/list.go:674: main.testListObjectsVersionsWithEmptyDirObject",
  "message": "ListObjectVersions returned unexpected listing result",
  "name": "versioning",
  "status": "FAIL"
}

Executed 10 out of 15 tests successfully.
```
<a name="b5d406bc-d1ae-46c5-8f7e-8d5b0bdd74fa"></a>
### 13941-f8a3fe9/mint-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s4:31778
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 5 seconds
(5/15) Running awscli tests ... done in 54 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 27 seconds
(8/15) Running minio-dotnet tests ... done in 51 seconds
(9/15) Running minio-go tests ... done in 1 minutes and 15 seconds
(10/15) Running minio-java tests ... FAILED in 1 minutes and 29 seconds
{
  "name": "minio-java",
  "function": "deleteUser()",
  "duration": 60004,
  "status": "FAIL",
  "error": "java.net.SocketTimeoutException: timeout >>> [okio.SocketAsyncTimeout.newTimeoutException(JvmOkio.kt:143), okio.AsyncTimeout.access$newTimeoutException(AsyncTimeout.kt:162), okio.AsyncTimeout$source$1.read(AsyncTimeout.kt:340), okio.RealBufferedSource.indexOf(RealBufferedSource.kt:449), okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.kt:333), okhttp3.internal.http1.HeadersReader.readLine(HeadersReader.kt:29), okhttp3.internal.http1.Http1ExchangeCodec.readResponseHeaders(Http1ExchangeCodec.kt:178), okhttp3.internal.connection.Exchange.readResponseHeaders(Exchange.kt:106), okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.kt:79), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:34), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201), okhttp3.internal.connection.RealCall.execute(RealCall.kt:154), io.minio.admin.MinioAdminClient.execute(MinioAdminClient.java:173), io.minio.admin.MinioAdminClient.deleteUser(MinioAdminClient.java:266), TestMinioAdminClient.deleteUser(TestMinioAdminClient.java:136), TestMinioAdminClient.runAdminTests(TestMinioAdminClient.java:148), FunctionalTest.runTests(FunctionalTest.java:3720), FunctionalTest.runEndpointTests(FunctionalTest.java:3818), FunctionalTest.main(FunctionalTest.java:3938)]"
}
(10/15) Running minio-js tests ... FAILED in 1 minutes and 24 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 17,
  "status": "FAIL",
  "error": "S3Error: Resource requested is unreadable, please reduce your request rate at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:89:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:164:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:88:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1334:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(10/15) Running minio-py tests ... FAILED in 6 minutes and 6 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-9704f284-a13f-4a93-87c3-85c763e51c78",
    "location": "default value ('us-east-1')"
  },
  "message": "S3 operation failed; code: BucketAlreadyOwnedByYou, message: Your previous request to create the named bucket succeeded and you already own it., resource: /minio-py-test-9704f284-a13f-4a93-87c3-85c763e51c78, request_id: 16C1A1B32802ABAD, host_id: 599bd4dc-a8ca-4d19-a3dc-48c6c21df46a, bucket_name: minio-py-test-9704f284-a13f-4a93-87c3-85c763e51c78",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 168, in test_make_bucket_default_region\n    _call(log_entry, _CLIENT.make_bucket, bucket_name)\n  File \"/mint/run/core/minio-py/tests.py\", line 109, in _call\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 612, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 380, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketAlreadyOwnedByYou, message: Your previous request to create the named bucket succeeded and you already own it., resource: /minio-py-test-9704f284-a13f-4a93-87c3-85c763e51c78, request_id: 16C1A1B32802ABAD, host_id: 599bd4dc-a8ca-4d19-a3dc-48c6c21df46a, bucket_name: minio-py-test-9704f284-a13f-4a93-87c3-85c763e51c78\n",
  "function": "make_bucket(bucket_name, location, object_lock)",
  "duration": 365987
}
(10/15) Running s3cmd tests ... FAILED in 3 minutes and 10 seconds
{
  "name": "s3cmd",
  "duration": "170947",
  "function": "test_put_object",
  "status": "FAIL",
  "error": "WARNING: Upload failed: /s3cmd-test-object-28976 (503 (SlowDown): Resource requested is unreadable, please reduce your request rate)\nWARNING: Retrying on lower speed (throttle=0.01)\nWARNING: Waiting 3 sec...\nWARNING: Upload failed: /s3cmd-test-object-28976 (503 (SlowDown): Resource requested is unreadable, please reduce your request rate)\nWARNING: Retrying on lower speed (throttle=0.05)\nWARNING: Waiting 6 sec...\nWARNING: Upload failed: /s3cmd-test-object-28976 (503 (SlowDown): Resource requested is unreadable, please reduce your request rate)\nWARNING: Retrying on lower speed (throttle=0.25)\nWARNING: Waiting 9 sec...\nWARNING: Upload failed: /s3cmd-test-object-28976 (503 (SlowDown): Resource requested is unreadable, please reduce your request rate)\nWARNING: Retrying on lower speed (throttle=1.25)\nWARNING: Waiting 12 sec...\nWARNING: Upload failed: /s3cmd-test-object-28976 (503 (SlowDown): Resource requested is unreadable, please reduce your request rate)\nWARNING: Retrying on lower speed (throttle=6.25)\nWARNING: Waiting 15 sec...\nWARNING: Too many failures. Giving up on '/mint/data/datafile-1-MB'\nERROR: Upload of '/mint/data/datafile-1-MB' failed too many times (Last reason: Too many failures. Giving up on '/mint/data/datafile-1-MB')"
}
(10/15) Running s3select tests ... FAILED in 9 minutes and 17 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "bucket_name": "s3select-test-912c24e4-9094-4c37-a252-68ca89674716"
  },
  "duration": 556771,
  "message": "S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /s3select-test-912c24e4-9094-4c37-a252-68ca89674716, request_id: 16C1A2617944ADCB, host_id: 599bd4dc-a8ca-4d19-a3dc-48c6c21df46a, bucket_name: s3select-test-912c24e4-9094-4c37-a252-68ca89674716",
  "error": "Traceback (most recent call last):\n  File \"./tests.py\", line 53, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 120, in test_csv_input_custom_quote_char\n    client.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 669, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 397, in _execute\n    return self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 380, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /s3select-test-912c24e4-9094-4c37-a252-68ca89674716, request_id: 16C1A2617944ADCB, host_id: 599bd4dc-a8ca-4d19-a3dc-48c6c21df46a, bucket_name: s3select-test-912c24e4-9094-4c37-a252-68ca89674716\n",
  "status": "FAIL"
}
(10/15) Running versioning tests ... done in 16 minutes and 51 seconds

Executed 10 out of 15 tests successfully.
```
<a name="8bcf9998-84f5-49a9-a745-689a02471672"></a>
### 13941-f8a3fe9/mint-pools.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s6:30864
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 2 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 44 seconds
(4/15) Running aws-sdk-ruby tests ... done in 5 seconds
(5/15) Running awscli tests ... done in 55 seconds
(6/15) Running healthcheck tests ... done in 1 seconds
(7/15) Running mc tests ... done in 28 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 6 seconds
(9/15) Running minio-go tests ... done in 1 minutes and 19 seconds
(10/15) Running minio-java tests ... FAILED in 1 minutes and 34 seconds
{
  "name": "minio-java",
  "function": "deleteUser()",
  "duration": 60004,
  "status": "FAIL",
  "error": "java.net.SocketTimeoutException: timeout >>> [okio.SocketAsyncTimeout.newTimeoutException(JvmOkio.kt:143), okio.AsyncTimeout.access$newTimeoutException(AsyncTimeout.kt:162), okio.AsyncTimeout$source$1.read(AsyncTimeout.kt:340), okio.RealBufferedSource.indexOf(RealBufferedSource.kt:449), okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.kt:333), okhttp3.internal.http1.HeadersReader.readLine(HeadersReader.kt:29), okhttp3.internal.http1.Http1ExchangeCodec.readResponseHeaders(Http1ExchangeCodec.kt:178), okhttp3.internal.connection.Exchange.readResponseHeaders(Exchange.kt:106), okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.kt:79), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:34), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201), okhttp3.internal.connection.RealCall.execute(RealCall.kt:154), io.minio.admin.MinioAdminClient.execute(MinioAdminClient.java:173), io.minio.admin.MinioAdminClient.deleteUser(MinioAdminClient.java:266), TestMinioAdminClient.deleteUser(TestMinioAdminClient.java:136), TestMinioAdminClient.runAdminTests(TestMinioAdminClient.java:148), FunctionalTest.runTests(FunctionalTest.java:3720), FunctionalTest.runEndpointTests(FunctionalTest.java:3818), FunctionalTest.main(FunctionalTest.java:3938)]"
}
(10/15) Running minio-js tests ... done in 1 minutes and 4 seconds
(11/15) Running minio-py tests ... done in 2 minutes and 25 seconds
(12/15) Running s3cmd tests ... done in 24 seconds
(13/15) Running s3select tests ... done in 8 seconds
(14/15) Running versioning tests ... FAILED in 39 minutes and 57 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-ka3habobq3e4sk",
    "expiry": 60000000000,
    "objectName": "testObject"
  },
  "duration": 2394037,
  "function": "/mint/build/versioning/main.go:79: main.cleanupBucket",
  "message": "Unable to cleanup bucket after compliance tests",
  "name": "versioning",
  "status": "FAIL"
}

Executed 13 out of 15 tests successfully.
```
<a name="9a745339-25e7-400b-afe5-8b8a5f8e0548"></a>
### 13941-f8a3fe9/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s3:31452
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 5 seconds
(5/15) Running awscli tests ... done in 1 minutes and 53 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 24 seconds
(8/15) Running minio-dotnet tests ... done in 55 seconds
(9/15) Running minio-go tests ... done in 1 minutes and 14 seconds
(10/15) Running minio-java tests ... FAILED in 1 minutes and 31 seconds
{
  "name": "minio-java",
  "function": "deleteUser()",
  "duration": 60004,
  "status": "FAIL",
  "error": "java.net.SocketTimeoutException: timeout >>> [okio.SocketAsyncTimeout.newTimeoutException(JvmOkio.kt:143), okio.AsyncTimeout.access$newTimeoutException(AsyncTimeout.kt:162), okio.AsyncTimeout$source$1.read(AsyncTimeout.kt:340), okio.RealBufferedSource.indexOf(RealBufferedSource.kt:449), okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.kt:333), okhttp3.internal.http1.HeadersReader.readLine(HeadersReader.kt:29), okhttp3.internal.http1.Http1ExchangeCodec.readResponseHeaders(Http1ExchangeCodec.kt:178), okhttp3.internal.connection.Exchange.readResponseHeaders(Exchange.kt:106), okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.kt:79), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:34), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201), okhttp3.internal.connection.RealCall.execute(RealCall.kt:154), io.minio.admin.MinioAdminClient.execute(MinioAdminClient.java:173), io.minio.admin.MinioAdminClient.deleteUser(MinioAdminClient.java:266), TestMinioAdminClient.deleteUser(TestMinioAdminClient.java:136), TestMinioAdminClient.runAdminTests(TestMinioAdminClient.java:148), FunctionalTest.runTests(FunctionalTest.java:3720), FunctionalTest.runEndpointTests(FunctionalTest.java:3818), FunctionalTest.main(FunctionalTest.java:3938)]"
}
(10/15) Running minio-js tests ... done in 57 seconds
(11/15) Running minio-py tests ... FAILED in 10 minutes and 11 seconds
{
  "name": "minio-py:test_put_object",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-f157e72b-5822-4ff7-b209-fa0de9d199fe",
    "object_name": "8c1d28ee-45ed-4c48-8aca-6c544545b034-metadata",
    "length": 11534336,
    "data": "LimitedRandomReader(11 * MB)",
    "metadata": {
      "x-amz-meta-testing": "value",
      "test-key": "value2"
    },
    "content_type": "application/octet-stream"
  },
  "message": "HTTPConnectionPool(host='minio-k8s3', port=31452): Max retries exceeded with url: /minio-py-test-f157e72b-5822-4ff7-b209-fa0de9d199fe (Caused by ResponseError('too many 503 error responses'))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 718, in test_put_object\n    _CLIENT.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 669, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 397, in _execute\n    return self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 846, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 846, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 846, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 836, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 574, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-k8s3', port=31452): Max retries exceeded with url: /minio-py-test-f157e72b-5822-4ff7-b209-fa0de9d199fe (Caused by ResponseError('too many 503 error responses'))\n",
  "duration": 606330
}
(11/15) Running s3cmd tests ... FAILED in 31 seconds
{
  "name": "s3cmd",
  "duration": "7632",
  "function": "test_put_object_multipart",
  "status": "FAIL",
  "error": "WARNING: Retrying failed request: /s3cmd-test-object-7990?uploadId=f11cde4b-9f6e-4ec6-8784-d3791a166630 (503 (SlowDown): Resource requested is unreadable, please reduce your request rate)\nWARNING: Waiting 3 sec...\nERROR: S3 error: 404 (NoSuchUpload): The specified multipart upload does not exist. The upload ID may be invalid, or the upload may have been aborted or completed."
}
(11/15) Running s3select tests ... FAILED in 5 minutes and 10 seconds
{
  "name": "s3select:test_sql_select",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "total_tests": 8,
    "total_success": 7
  },
  "duration": 300801,
  "message": "S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /s3select-test-8f7858f3-c715-43d4-886a-af3dc56e0283, request_id: 16C1A232DF5B36CE, host_id: 17679654-5b4e-4b7c-b2e0-3d3c1e0186a4, bucket_name: s3select-test-8f7858f3-c715-43d4-886a-af3dc56e0283",
  "error": "Traceback (most recent call last):\n  File \"./tests.py\", line 84, in main\n    test_sql_select(client, log_output)\n  File \"/mint/run/core/s3select/sql_ops.py\", line 369, in test_sql_select\n    raise select_err\n  File \"/mint/run/core/s3select/sql_ops.py\", line 367, in test_sql_select\n    test_sql_expressions(client, json_testfile, tests, log_output)\n  File \"/mint/run/core/s3select/sql_ops.py\", line 90, in test_sql_expressions\n    test_sql_expressions_custom_input_output(client, input_json_bytes,\n  File \"/mint/run/core/s3select/sql_ops.py\", line 78, in test_sql_expressions_custom_input_output\n    client.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 669, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 397, in _execute\n    return self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 380, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /s3select-test-8f7858f3-c715-43d4-886a-af3dc56e0283, request_id: 16C1A232DF5B36CE, host_id: 17679654-5b4e-4b7c-b2e0-3d3c1e0186a4, bucket_name: s3select-test-8f7858f3-c715-43d4-886a-af3dc56e0283\n",
  "status": "FAIL"
}
(11/15) Running versioning tests ... done in 28 minutes and 4 seconds

Executed 11 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/admin-handlers-users.go
+++ b/cmd/admin-handlers-users.go
@@ -58,17 +58,9 @@ func (a adminAPIHandlers) RemoveUser(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if err := globalIAMSys.DeleteUser(ctx, accessKey); err != nil {
+	if err := globalIAMSys.DeleteUser(ctx, accessKey, true); err != nil {
 		writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
 		return
-	}
-
-	// Notify all other MinIO peers to delete user.
-	for _, nerr := range globalNotificationSys.DeleteUser(accessKey) {
-		if nerr.Err != nil {
-			logger.GetReqInfo(ctx).SetTags("peerAddress", nerr.Host.String())
-			logger.LogIf(ctx, nerr.Err)
-		}
 	}
 }
 
@@ -978,16 +970,10 @@ func (a adminAPIHandlers) DeleteServiceAccount(w http.ResponseWriter, r *http.Re
 		return
 	}
 
-	err = globalIAMSys.DeleteServiceAccount(ctx, serviceAccount)
+	err = globalIAMSys.DeleteServiceAccount(ctx, serviceAccount, true)
 	if err != nil {
 		writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
 		return
-	}
-	for _, nerr := range globalNotificationSys.DeleteServiceAccount(serviceAccount) {
-		if nerr.Err != nil {
-			logger.GetReqInfo(ctx).SetTags("peerAddress", nerr.Host.String())
-			logger.LogIf(ctx, nerr.Err)
-		}
 	}
 
 	// Call site replication hook. Only LDAP accounts are supported for

--- a/cmd/iam.go
+++ b/cmd/iam.go
@@ -536,12 +536,26 @@ func (sys *IAMSys) SetPolicy(ctx context.Context, policyName string, p iampolicy
 }
 
 // DeleteUser - delete user (only for long-term users not STS users).
-func (sys *IAMSys) DeleteUser(ctx context.Context, accessKey string) error {
+func (sys *IAMSys) DeleteUser(ctx context.Context, accessKey string, notifyPeers bool) error {
 	if !sys.Initialized() {
 		return errServerNotInitialized
 	}
 
-	return sys.store.DeleteUser(ctx, accessKey, regUser)
+	if err := sys.store.DeleteUser(ctx, accessKey, regUser); err != nil {
+		return err
+	}
+
+	// Notify all other MinIO peers to delete user.
+	if notifyPeers && !sys.HasWatcher() {
+		for _, nerr := range sys.notificationSys.DeleteUser(accessKey) {
+			if nerr.Err != nil {
+				logger.GetReqInfo(ctx).SetTags("peerAddress", nerr.Host.String())
+				logger.LogIf(ctx, nerr.Err)
+			}
+		}
+	}
+
+	return nil
 }
 
 // CurrentPolicies - returns comma separated policy string, from
@@ -912,7 +926,7 @@ func (sys *IAMSys) GetClaimsForSvcAcc(ctx context.Context, accessKey string) (ma
 }
 
 // DeleteServiceAccount - delete a service account
-func (sys *IAMSys) DeleteServiceAccount(ctx context.Context, accessKey string) error {
+func (sys *IAMSys) DeleteServiceAccount(ctx context.Context, accessKey string, notifyPeers bool) error {
 	if !sys.Initialized() {
 		return errServerNotInitialized
 	}
@@ -922,7 +936,20 @@ func (sys *IAMSys) DeleteServiceAccount(ctx context.Context, accessKey string) e
 		return nil
 	}
 
-	return sys.store.DeleteUser(ctx, accessKey, svcUser)
+	if err := sys.store.DeleteUser(ctx, accessKey, svcUser); err != nil {
+		return err
+	}
+
+	if notifyPeers && !sys.HasWatcher() {
+		for _, nerr := range sys.notificationSys.DeleteServiceAccount(accessKey) {
+			if nerr.Err != nil {
+				logger.GetReqInfo(ctx).SetTags("peerAddress", nerr.Host.String())
+				logger.LogIf(ctx, nerr.Err)
+			}
+		}
+	}
+
+	return nil
 }
 
 // CreateUser - create new user credentials and policy, if user already exists

--- a/cmd/peer-rest-server.go
+++ b/cmd/peer-rest-server.go
@@ -157,7 +157,7 @@ func (s *peerRESTServer) DeleteServiceAccountHandler(w http.ResponseWriter, r *h
 		return
 	}
 
-	if err := globalIAMSys.DeleteServiceAccount(r.Context(), accessKey); err != nil {
+	if err := globalIAMSys.DeleteServiceAccount(r.Context(), accessKey, false); err != nil {
 		s.writeErrorResponse(w, err)
 		return
 	}
@@ -209,7 +209,7 @@ func (s *peerRESTServer) DeleteUserHandler(w http.ResponseWriter, r *http.Reques
 		return
 	}
 
-	if err := globalIAMSys.DeleteUser(r.Context(), accessKey); err != nil {
+	if err := globalIAMSys.DeleteUser(r.Context(), accessKey, false); err != nil {
 		s.writeErrorResponse(w, err)
 		return
 	}

--- a/cmd/site-replication.go
+++ b/cmd/site-replication.go
@@ -1080,16 +1080,9 @@ func (c *SiteReplicationSys) PeerSvcAccChangeHandler(ctx context.Context, change
 		}
 
 	case change.Delete != nil:
-		err := globalIAMSys.DeleteServiceAccount(ctx, change.Delete.AccessKey)
+		err := globalIAMSys.DeleteServiceAccount(ctx, change.Delete.AccessKey, true)
 		if err != nil {
 			return wrapSRErr(err)
-		}
-
-		for _, nerr := range globalNotificationSys.DeleteServiceAccount(change.Delete.AccessKey) {
-			if nerr.Err != nil {
-				logger.GetReqInfo(ctx).SetTags("peerAddress", nerr.Host.String())
-				logger.LogIf(ctx, nerr.Err)
-			}
 		}
 
 	}

